types/gestalt/index.d.ts

```diff
@@ -885,7 +882,7 @@ export interface ModalProps {
      */
     closeOnOutsideClick?: boolean;
     footer?: React.ReactNode;
-    heading?: React.ReactNode;
+    heading?: React.ReactNode | string;
```

Comment:
nit: `React.ReactNode` includes `string` type so `React.ReactNode | string` is redundant.